src/shogun/machine/GeneralizedLinearMachine.cpp

```diff
@@ -0,0 +1,299 @@
+/*
+ * STILL A WORK IN PROGRESS
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Tej Sukhatme
+ */
+
+// #include <rxcpp/rx-lite.hpp>
+// #include <shogun/features/DotFeatures.h>
+// #include <shogun/labels/Labels.h>
+// #include <shogun/labels/RegressionLabels.h>
+#include <shogun/machine/GeneralizedLinearMachine.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+// #include <utility>
+#include <math.h>
+#include <random>
+
+using namespace shogun;
+
+GeneralizedLinearMachine::GeneralizedLinearMachine(): Machine()
+{
+	init();
+}
+
+GeneralizedLinearMachine::GeneralizedLinearMachine(const std::shared_ptr<GeneralizedLinearMachine>& machine) : Machine()
+{
+	init();
+	require(machine, "No machine provided.");
+
+	auto w = machine->get_w();
+	auto w_clone = w.clone();
+	set_w(w_clone);
+	set_bias(machine->get_bias());
+}
+
+void GeneralizedLinearMachine::init()
+{
+	m_bias = 0;
+	features = NULL;
+	m_w = NULL;
+
+	distribution= POISSON;
+	m_alpha=0.5;
+	m_tau=NULL;
+	m_lambda=0.1;
+	m_learning_rate=2e-1;
```

Comment:
i think it might be a good idea to parametrize the gradient descent method, using methods in shogun (which would then store the learning rate)

Reply:
and generally the optimization variables